perf_test/KokkosKernels_perf_test_instantiation.hpp

```diff
@@ -0,0 +1,133 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+//
+// Created by Berger-Vergiat, Luc on 2/6/23.
+//
+
+#ifndef KOKKOSKERNELS_PERF_TEST_INSTANTIATION_HPP
+#define KOKKOSKERNELS_PERF_TEST_INSTANTIATION_HPP
+
+#include "KokkosKernels_perf_test_utilities.hpp"
+
+#ifndef KOKKOSKERNELS_PERF_TEST_NAME
+#error "The macro KOKKOSKERNELS_PERF_TEST_NAME was not defined"
+#endif
+
+int main_instantiation(int argc, char** argv) {
+  perf_test::CommonInputParams params;
+  perf_test::parse_common_options(argc, argv, params);
+
+  /* Assumption is that use_openmp/use_threads variables are */
+  /* provided as numbers of threads */
+  int num_threads = 1;
+  if (params.use_openmp) {
+    num_threads = params.use_openmp;
+  } else if (params.use_threads) {
+    num_threads = params.use_threads;
+  }
+
+  int device_id = 0;
+  if (params.use_cuda)
+    device_id = params.use_cuda - 1;
+  else if (params.use_hip)
+    device_id = params.use_hip - 1;
+  else if (params.use_sycl)
+    device_id = params.use_sycl - 1;
+
+  Kokkos::initialize(Kokkos::InitializationSettings()
+                         .set_num_threads(num_threads)
+                         .set_device_id(device_id));
+  Kokkos::print_configuration(std::cout);
+  std::cout << '\n';
+
+  bool ran = false;
+
+  if (params.use_openmp) {
+#if defined(KOKKOS_ENABLE_OPENMP)
+    std::cout << "Running on OpenMP backend.\n";
+    KOKKOSKERNELS_PERF_TEST_NAME<Kokkos::OpenMP>(argc, argv, params);
```

Comment:
See https://github.com/kokkos/kokkos-kernels/blob/master/perf_test/blas/blas3/KokkosBlas3_gemm_perf_test.hpp#L73-L133 for another approach using a table of function pointers.